src/terms/anyonic.jl

```diff
@@ -22,17 +68,36 @@ function (A::Anyonic)(basis)
     @assert basis.model.lattice[2, 1] == basis.model.lattice[1, 2] == 0
     @assert basis.model.lattice[1, 1] == basis.model.lattice[2, 2]
 
-    TermAnyonic(basis, eltype(basis)(A.β))
+    TermAnyonic(basis, eltype(basis)(A.hbar), eltype(basis)(A.β))
 end
 
-struct TermAnyonic{T <: Real} <: Term
+struct TermAnyonic{T <: Real, Tρ <: RealFourierArray, TA} <: Term
     basis::PlaneWaveBasis{T}
+    hbar::T
     β::T
+    ρref::Tρ
+    Aref::TA
+end
+function TermAnyonic(basis::PlaneWaveBasis{T}, hbar, β) where {T}
+    # compute correction magnetic field
+    ρref = zeros(T, basis.fft_size)
+    Aref = [zeros(T, basis.fft_size), zeros(T, basis.fft_size)]
+    M = basis.model.n_electrons  # put M to zero to disable
+    σ = 2
+    for (ir, r) in enumerate(r_vectors(basis))
+        rcart = basis.model.lattice * (r - @SVector[.5, .5, .0])
+        ρref[ir] = ρref_real(rcart[1], rcart[2], M, σ)
+        Aref[1][ir], Aref[2][ir] = magnetic_field_produced_by_ρref(rcart[1], rcart[2], M, σ)
```

Comment:
Line too long.

Reply:
I generally agree with your line breaks when it helps readability by breaking up complex expressions, but here there's no natural way to break

Reply:
```
res = magnetic_field_produced_by_ρref(rcart[1], rcart[2], M, σ)
Aref[1][ir], Aref[2][ir] = res
```

Reply:
you can't seriously think this is more readable than the previous version?

Reply:
I think it is, but fair enough.

Reply:
I'm trying, but I just can't understand your objection to long lines. I've seen your terminal, you have enough space to display them, and so does everybody I've seen. I do agree that long lines are very often a sign of bad code, like long functions and a large number of indentation levels, but it's not bad *per se*, is it? Like here, it's just because the function name is very long

Reply:
The part where it is bad per se is that it makes it harder to read independent from the fact that it is code.

Basically it's the same reason why papers and books are typically printed in two columns if they are on A4. When lines get too long the eye can't follow and you have to read the same thing more often or you miss part of the info. The recommended number of characters on a line is something between 70 and 90. (Arguably this is 92, so it's actually ok, but we should really aim below 90 chars to make sure that it does not become 150 in the end.)

Reply:
Also my experience is that accepting long lines sometimes stops one too early from trying to make code formatted in a nicer way.

Reply:
This is hugely different for unstructured text (where the unit is the word or sentence and lines have no specific meaning) and for imperative code (where the unit is the line of code). In the case of code, it is much more important to preserve the integrity of the line of code (like in the example above where breaking the line breaks the natural flow of the assignment). What you're saying could plausibly be true for more functional languages which are less linear in their form, but not for us.

Reply:
Yeah I agree with the fact that long lines are a sign to be suspicious, but it doesn't mean it's unconditionally bad

Reply:
If the get too long they are unconditionally bad. Probably not in this example though, that I can agree upon.